test/feeBurner.js

```diff
@@ -644,9 +640,96 @@ contract('FeeBurner', function(accounts) {
             assert(Helper.isRevertErrorMessage(e), "expected throw but got other error: " + e);
         }
     });
-});
 
+    it("should verify when have knc arbitrage, set rate reverts.", async function() {
+        aRate = 1.2345;
+        const ethToKncRatePrecision = precisionUnits.mul(aRate * 1.01);
+        const kncToEthRatePrecision = precisionUnits.div(aRate);
+
+        await mockKyberNetwork.setPairRate(ethAddress, kncToken.address, ethToKncRatePrecision);
+        await mockKyberNetwork.setPairRate(kncToken.address, ethAddress, kncToEthRatePrecision);
+
+        let rxEthToKnc = await mockKyberNetwork.getExpectedRate(ethAddress, kncToken.address, 10 ** 19);
+        let rxKncToEth = await mockKyberNetwork.getExpectedRate(kncToken.address, ethAddress, 10 ** 19);
+        assert.equal(rxEthToKnc[0].valueOf(), ethToKncRatePrecision.floor().valueOf())
+        assert.equal(rxKncToEth[0].valueOf(), kncToEthRatePrecision.floor().valueOf())
+
+        let burnerInitialkncRate = await feeBurnerInst.kncPerEthRatePrecision();
+//         has arb. get rate will return 0. should revert.
+        try {
+            let rxx = await feeBurnerInst.setKNCRate();
+            assert(false, "throw was expected in line above.")
+        } catch(e){
+            assert(Helper.isRevertErrorMessage(e), "expected throw but got: " + e);
+        }
+
+        let burnerkncRateAfterSet = await feeBurnerInst.kncPerEthRatePrecision();
+        assert.equal(burnerkncRateAfterSet.valueOf(), burnerInitialkncRate.valueOf());
+      });
+
+      it("should verify when have no knc arbitrage, set rate success", async function() {
+        aRate = 1.2345;
+        const ethToKncRatePrecision = precisionUnits.mul(aRate * 1.01);
+        const kncToEthRatePrecision = precisionUnits.div(aRate);
 
+        await mockKyberNetwork.setPairRate(ethAddress, kncToken.address, ethToKncRatePrecision);
+        await mockKyberNetwork.setPairRate(kncToken.address, ethAddress, kncToEthRatePrecision);
+
+        let rxEthToKnc = await mockKyberNetwork.getExpectedRate(ethAddress, kncToken.address, 10 ** 19);
+        let rxKncToEth = await mockKyberNetwork.getExpectedRate(kncToken.address, ethAddress, 10 ** 19);
+        assert.equal(rxEthToKnc[0].valueOf(), ethToKncRatePrecision.floor().valueOf())
+        assert.equal(rxKncToEth[0].valueOf(), kncToEthRatePrecision.floor().valueOf())
+
```

Comment:
can test here failing rate before setting non failing rate (no arb)

Reply:
As mentioned above.

Reply:
OK
lets keep it.